pyhive/hive.py

```diff
@@ -299,23 +302,29 @@ def fetch_logs(self):
         if self._state == self._STATE_NONE:
             raise ProgrammingError("No query yet")
 
-        logs = []
-        while True:
-            req = ttypes.TFetchResultsReq(
-                operationHandle=self._operationHandle,
-                orientation=ttypes.TFetchOrientation.FETCH_NEXT,
-                maxRows=self.arraysize,
-                fetchType=1  # 0: results, 1: logs
-            )
-            response = self._connection.client.FetchResults(req)
-            _check_status(response)
-            assert not response.results.rows, 'expected data in columnar format'
-            assert len(response.results.columns) == 1, response.results.columns
-            new_logs = _unwrap_column(response.results.columns[0])
-            logs += new_logs
+        try:  # Older Hive instances require logs to be retrieved using GetLog
```

Comment:
I think the reason it was done in this order is because the error syndrome when attempting to collected logs using new style method is somewhat hard to distinguish from other pathologies. In particular, it basically returns no data and says that the status is still "RUNNING". One could imagine that reflecting other pathologies, whereas the error when trying the old method first is very clear: "UNKNOWN_METHOD".

As for which versions, that's hard to say exactly. But I know it affects Cloudera based Hive 0.13. I suspect it only affects Impala systems, but I'm not sure.